digitiser-aggregator/src/frame/aggregated.rs

```diff
@@ -1,45 +1,45 @@
-use super::partial::{PartialFrame, PartialFrameLike};
+use super::partial::PartialFrame;
 use crate::data::{Accumulate, DigitiserData};
-use std::fmt::Debug;
+use supermusr_common::spanned::{SpanOnce, Spanned, SpannedMut};
 #[cfg(test)]
 use supermusr_common::DigitizerId;
 use supermusr_streaming_types::FrameMetadata;
 
-pub(crate) trait AggregatedFrameLike<D, P: PartialFrameLike<D>>:
-    Debug + AsRef<AggregatedFrame<D>> + AsMut<AggregatedFrame<D>> + From<P>
-{
-}
-
-impl<D> AsRef<Self> for AggregatedFrame<D> {
-    fn as_ref(&self) -> &Self {
-        self
-    }
-}
-impl<D> AsMut<Self> for AggregatedFrame<D> {
-    fn as_mut(&mut self) -> &mut Self {
-        self
-    }
-}
-
-impl<D: Debug> AggregatedFrameLike<D, PartialFrame<D>> for AggregatedFrame<D> where
-    Vec<(u8, D)>: Accumulate<D>
-{
-}
-
-#[derive(Debug)]
 pub(crate) struct AggregatedFrame<D> {
+    span: SpanOnce,
     pub(crate) metadata: FrameMetadata,
     #[cfg(test)]
     pub(crate) digitiser_ids: Vec<DigitizerId>,
     pub(crate) digitiser_data: D,
 }
 
+#[cfg(test)]
+impl<D> AggregatedFrame<D> {
+    pub(crate) fn new(
+        metadata: FrameMetadata,
+        #[cfg(test)] digitiser_ids: Vec<DigitizerId>,
+        digitiser_data: D,
+    ) -> Self {
+        Self {
+            span: Default::default(),
+            metadata,
+            #[cfg(test)]
+            digitiser_ids,
+            digitiser_data,
+        }
+    }
+}
+
 impl<D> From<PartialFrame<D>> for AggregatedFrame<D>
 where
     DigitiserData<D>: Accumulate<D>,
 {
     fn from(mut partial: PartialFrame<D>) -> Self {
         Self {
+            span: partial
+                .span_mut()
+                .take()
```

Comment:
If I am reading `spanned.rs` correctly then this should only give an error in the event of a logic error in the code, not some unknown external factor, correct?

In which case simply using `unwrap()` is fine.

At some point in the future it might be worth revisiting the error handing options, which I'll describe in https://github.com/STFC-ICD-Research-and-Design/supermusr-data-pipeline/issues/161

Reply:
Yes, it's just meant to enforce proper use of the SpanOnce type.